include/pagmo/algorithms/pso_gen.hpp

```diff
@@ -0,0 +1,881 @@
+        stream(ss, "\n\tVariant: ", m_variant);
+        stream(ss, "\n\tTopology: ", m_neighb_type);
+        if (m_neighb_type == 2u || m_neighb_type == 4u) {
+            stream(ss, "\n\tTopology parameter: ", m_neighb_param);
+        }
+        stream(ss, "\n\tMemory: ", m_memory);
+        stream(ss, "\n\tSeed: ", m_seed);
+        stream(ss, "\n\tVerbosity: ", m_verbosity);
+        return ss.str();
+    }
+    /// Get log
+    /**
+     * A log containing relevant quantities monitoring the last call to evolve. Each element of the returned
+     * <tt>std::vector</tt> is a pso_gen::log_line_type containing: Gen, Fevals, gbest,
+     * Mean Vel., Mean lbest, Avg. Dist. as described in pso_gen::set_verbosity
+     * @return an <tt>std::vector</tt> of pso_gen::log_line_type containing the logged values
+     */
+    const log_type &get_log() const
+    {
+        return m_log;
+    }
+    /// Object serialization
+    /**
+     * This method will save/load \p this into the archive \p ar.
+     *
+     * @param ar target archive.
+     *
+     * @throws unspecified any exception thrown by the serialization of the UDA and of primitive types.
+     */
+    template <typename Archive>
+    void serialize(Archive &ar)
+    {
+        ar(m_max_gen, m_omega, m_eta1, m_eta2, m_max_vel, m_variant, m_neighb_type, m_neighb_param, m_e, m_seed,
+           m_verbosity, m_log);
+    }
+
+private:
+    /**
+     *  @brief Get information on the best position already visited by any of a particle's neighbours
+     *
+     *  @param pidx index to the particle under consideration
+     *  @param neighb definition of the swarm's topology
+     *  @param lbX particles' previous best positions
+     *  @param lbfit particles' fitness values at their previous best positions
+     *  @return best position already visited by any of the considered particle's neighbours
+     */
+    vector_double particle__get_best_neighbor(population::size_type pidx,
+                                              std::vector<std::vector<vector_double::size_type>> &neighb,
+                                              const std::vector<vector_double> &lbX,
+                                              const std::vector<vector_double> &lbfit) const
+    {
+        population::size_type bnidx; // neighbour index; best neighbour index
+
+        switch (m_neighb_type) {
+            case 1: // { gbest }
+                // ERROR: execution should not reach this point, as the global best position is not tracked using the
+                // neighb vector
+                pagmo_throw(                                                                     // LCOV_EXCL_LINE
+                    std::invalid_argument,                                                       // LCOV_EXCL_LINE
+                    "particle__get_best_neighbor() invoked while using a gbest swarm topology"); // LCOV_EXCL_LINE
+                break;
+            case 2: // { lbest }
```

Comment:
Like above, can we remove these empty ``case``s or there is some other reason to keep them here?

Reply:
left it, see reason above.